api/provider/aws/service_delete.go

```diff
@@ -1,5 +1,65 @@
 package aws
 
-func (s *ServiceProvider) Delete(ServiceID string) error {
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+// Delete a specified service within a cluster.
+func (s *ServiceProvider) Delete(serviceID string) error {
+	desiredCount := int64(0)
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		return err
+	}
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+	fqClusterID := addLayer0Prefix(s.Config.Instance(), environmentID)
+
+	if err := s.scaleService(fqServiceID, fqClusterID, desiredCount); err != nil {
+		return fmt.Errorf("Failed to scale service: %e", err)
+	}
+
+	if err := s.deleteService(fqServiceID, fqClusterID); err != nil {
+		return fmt.Errorf("Failed to delete service: %e", err)
+	}
+
+	if err := deleteEntityTags(s.TagStore, "service", serviceID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) scaleService(serviceID, clusterID string, desiredCount int64) error {
```

Comment:
one of the "rules of thumb" is to hide ugliness caused by `aws-sdk-go` - using an `int64` where an `int` should be is a pretty common one.

Reply:
So should I just cast it to a int64 then for the sdk

Reply:
What we've been doing is to keep the types in the models, master functions, and non-AWS helper functions simple. Then we'd cast as needs require within the AWS helper functions as we build `input` objects.